### PR TITLE
3 create metadata table

### PR DESCRIPTION
**Pull request by q-w-a**
Code to parse directory of xml files.

The path assumes you have unzipped the data in the project directory. It will also generate a folder in the current directory named `data` with the output file; this folder is ignored via `gitignore`.

### CODE REVIEW

**Comment by q-w-a** [comment]
Currently working on finishing this task; I have some more commits to add before it'll be ready to merge.

File: generate_table.Rmd [hunk 1]
```diff
@@ -0,0 +1,624 @@
...
+  rmdformats::readthedown:
+    df_print: paged
+    code_folding: hide
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE, message=FALSE, warning=FALSE)
+```
+
+```{css,echo=FALSE}
```
Comment (raevard) [inline generate_table.Rmd:15]: I'm assuming the CSS code here is for formatting knitted HTMLS; could you add some comments to confirm?

File: generate_table.Rmd [hunk 2]
```diff
@@ -0,0 +1,624 @@
...
+
+# check all files are present in data frame
+# nrow(all_data) == length(files)
+
+saveRDS(all_data, "./data/data_990.RDS")
+
+```
+
+
+```{r troubleshooting, include=FALSE,eval=FALSE}
```
Comment (raevard) [inline generate_table.Rmd:205]: To confirm, this was your test indicating that you needed to handle the special case of IRS, which you've done, so now the comment in this chunk isn't accurate?

File: generate_table.Rmd [hunk 3]
```diff
@@ -0,0 +1,624 @@
...
+     }
+   }
+   
+ # add name of file to data frame
+ extracted %>%
+   mutate(filename = gsub("*.*\\/", "", filename))
+}
+
+
+
```
Comment (raevard) [inline generate_table.Rmd:179]: Can you add a comment or two more here to indicate what it's doing?  From what I can tell it is: 
- Creating a new folder ./data
- Applying get_df to all files with the earlier specified variables to extract
- Wrangling all-data to have proper formatting and new variables like fiscal_year
- Saving the wrangled table for future use

File: generate_table.Rmd [hunk 4]
```diff
@@ -0,0 +1,624 @@
...
+library(xml2)
+library(kableExtra)
+library(lubridate)
+library(forcats)
+
+
+```
+
+# Generating Table for All Files
+
```
Comment (raevard) [inline generate_table.Rmd:64]: Absolutely fantastic work done here, Quinn!!! The only comments I include are for readability of code. Suuuuper minor, I'm wondering if this initial chunk could be rearranged a bit, so the function definition is by itself, then the variable creation (files, variables_to_extract) and applying get_df all happen in the same place?

**Review by raevard — APPROVED** [review]

**Pull request merged**